chain/network/src/peer_manager/connection_store/tests.rs

```diff
@@ -0,0 +1,128 @@
+use crate::peer_manager::connection_store::ConnectionStore;
+use crate::peer_manager::connection_store::OUTBOUND_CONNECTIONS_CACHE_SIZE;
+use crate::store;
+use crate::time;
+use crate::types::{ConnectionInfo, PeerInfo};
+use near_crypto::{KeyType, SecretKey};
+use near_primitives::network::PeerId;
+use near_store::NodeStorage;
+use rand::Rng;
+use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4};
+
+fn get_addr(port: u16) -> SocketAddr {
+    SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), port).into()
+}
+
+fn gen_peer_info(port: u16) -> PeerInfo {
+    PeerInfo {
+        id: PeerId::new(SecretKey::from_random(KeyType::ED25519).public_key()),
+        addr: Some(get_addr(port)),
+        account_id: None,
+    }
+}
+
+fn check_outbound_connections(store: &ConnectionStore, wanted: &Vec<ConnectionInfo>) {
+    let got = store.get_recent_outbound_connections();
+    for c in &got {
+        assert!(wanted.contains(c));
+    }
+    for c in wanted {
+        assert!(got.contains(&*c));
+    }
+}
+
+#[test]
+fn test_store_connection() {
+    let clock = time::FakeClock::default();
+    let (_tmp_dir, opener) = NodeStorage::test_opener();
+
+    let now_utc = clock.now_utc();
+
+    let conn_info_a = ConnectionInfo {
+        peer_info: gen_peer_info(0),
+        time_established: now_utc - time::Duration::hours(1),
+        time_connected_until: now_utc,
+    };
+
+    {
+        let store = store::Store::from(opener.open().unwrap());
```

Comment:
why not near_store::db::TestDB::new() ?

Reply:
There is no need to test file reopening. It would be enough to pass the same store to the constructor again (after destroying the ConnectionStore instance).